api/middleware.ts

```diff
@@ -215,6 +216,17 @@ const checkNewPassword = function (field: string): ValidationChain {
   });
 };
 
+const viewMode = function(): ValidationChain {
+  // All api listing commands will automatically return all results if the user is a super-admin
+  // There is now an optional "view-mode" query param to these APIs, which, if set to "limited",
+  // will restrict results to items only directly viewable by the super-admin user.
+  // The default behaviour remains unchanged, and this will do nothing for non-admin users.
+  return query("view-mode").custom((value, {req}) => {
+    req.body.viewAsSuperAdmin = value !== "limited";
```

Comment:
I find it a bit confusing that it is a query test but then you are setting the value in the body.   Mostly I am just finding it confusing because it is magic!